### ISSUE TITLE
iOS 13 not working?

### ISSUE THREAD

**Issue by Richi2293** [react-native-community/react-native-image-picker#1179]
# Bug

Only on iOS 13 does not work and gives me this error, iOS 12 works properly.

## Environment info

React native info output:

```bash
React Native Environment Info:
    System:
      OS: macOS 10.14.5
      CPU: (4) x64 Intel(R) Core(TM) i5-4258U CPU @ 2.40GHz
      Memory: 629.88 MB / 8.00 GB
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 8.11.1 - /usr/local/bin/node
      Yarn: 1.17.3 - ~/.yarn/bin/yarn
      npm: 6.7.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 13.0, DriverKit 19.0, macOS 10.15, tvOS 13.0, watchOS 6.0
      Android SDK:
        API Levels: 19, 23, 24, 25, 26, 27, 28
        Build Tools: 23.0.1, 23.0.2, 23.0.3, 25.0.0, 25.0.1, 25.0.2, 25.0.3, 26.0.1, 26.0.2, 26.0.3, 27.0.0, 27.0.1, 27.0.2, 27.0.3, 28.0.0, 28.0.0, 28.0.1, 28.0.2, 28.0.3, 29.0.0
        System Images: android-25 | Google Play Intel x86 Atom, android-27 | Google Play Intel x86 Atom, android-28 | Google APIs Intel x86 Atom
    IDEs:
      Android Studio: 3.4 AI-183.6156.11.34.5692245
      Xcode: 11.0/11A420a - /usr/bin/xcodebuild
    npmPackages:
      react: 16.8.3 => 16.8.3
      react-native: 0.59.9 => 0.59.9
    npmGlobalPackages:
      create-react-native-app: 1.0.0
      react-native-cli: 2.0.1
      react-native-create-library: 3.1.2
      react-native-git-upgrade: 0.2.7
```

Library version: 1.1.0

## Steps To Reproduce

1. run code ImagePicker.showImagePicker for pick a video
2. wait callback response
3. message: 'Response = ', { error: 'You don’t have permission.' }

Describe what you expected to happen:

1. run code ImagePicker.showImagePicker for pick a video
2. wait callback response
3. The video has been uploaded correctly

## Reproducible sample code

Use the code in Readme under Usage.

**Comment by zirkc** [comment]
I fixed it by changing moveItemAtURL to copyItemAtURL at line 459 on ImagePickerManager.m

**Comment by matthamil** [comment]
We encountered an issue on iOS 13 and ended up discovering the same fix that @zirkc recommended. For us, changing `copyItemAtURL` to `moveItemAtURL` on that line resolves the issue. The tradeoff is that the original file is not being removed from its source location, which appears to be in a `/tmp` directory. I assume this `/tmp` directory will be periodically emptied though.

**Comment by creativetim** [comment]
Being _very_ new to iOS development, after finding the same fix, I'm left scratching my head and wondering _how much do I really know about file permissions re: move vs copy?_

**Comment by zirkc** [comment]
prob because you are not allowed to remove items from certain directories.

**Comment by daylytes-oakridge** [comment]
#1118

**Comment by rewieer** [comment]
For those interested in the patch-package file on 0.28.0 (react-native-image-picker+0.28.1.patch) :
```
diff --git a/node_modules/react-native-image-picker/ios/ImagePickerManager.m b/node_modules/react-native-image-picker/ios/ImagePickerManager.m
index 28d5870..8c060b1 100644
--- a/node_modules/react-native-image-picker/ios/ImagePickerManager.m
+++ b/node_modules/react-native-image-picker/ios/ImagePickerManager.m
@@ -456,7 +456,7 @@ - (void)imagePickerController:(UIImagePickerController *)picker didFinishPicking

                 if (videoURL) { // Protect against reported crash
                   NSError *error = nil;
-                  [fileManager moveItemAtURL:videoURL toURL:videoDestinationURL error:&error];
+                  [fileManager copyItemAtURL:videoURL toURL:videoDestinationURL error:&error];
                   if (error) {
                       self.callback(@[@{@"error": error.localizedFailureReason}]);
                       return;
```

**Comment by vincentzierigen** [comment]
please, merge it

**Comment by mdere-unbound** [comment]
hmm I changed it locally on my node module and it's still crashing for me - am I missing something?

**Comment by maxschmeling** [comment]
@mdere-unbound did you rebuild the app in Xcode after making the change?

**Comment by mattslight** [comment]
@mdere-unbound make sure you also follow the post install steps to include the necessary permissions in Info.plist

**Comment by lafiosca** [comment]
Is there some label that can be added to make stalebot stop on long-standing issues like this?

**Comment by Johan-dutoit** [comment]
Perhaps someone can submit a PR?

**Comment by lafiosca** [comment]
@Johan-dutoit Someone submitted a PR back in July, linked earlier in this thread, but none of the maintainers replied to it: #1118

**Comment by Johan-dutoit** [comment]
Looks like it was merged to me?  Which PR are you referring too?

**Comment by lafiosca** [comment]
Thank you, much appreciated!

**Comment by Johan-dutoit** [comment]
Should be fixed with [v2.2.1](https://github.com/react-native-community/react-native-image-picker/releases/tag/v2.2.1).

Closing for now, if the issue persists, then we can reopen.
